bundle/src/bundle_execution.rs

```diff
@@ -251,8 +251,22 @@ pub fn load_and_execute_bundle<'a>(
             metrics: BundleExecutionMetrics::default(),
         };
     }
+
     let mut binding = AccountOverrides::default();
-    let account_overrides = account_overrides.unwrap_or(&mut binding);
+    let account_overrides = if is_simulation {
+        bundle
```

Comment:
this changes logic, shouldn't you conditionally use account_overrides or binding?